File: client/src/app/core/models/contributionRequest.ts
export class ContributionRequest {
    name: string;
    nominal: boolean;
    values: { [key: number]: number };
    portfolio: number;

    constructor(key: number = 1) {
        this.name = '';
        this.nominal = false;
        this.values = { [key]: 0 };
        this.portfolio = 0;
    }
}

File: Simulation/Simulation.Domain/Entities/ContributionRequest.cs
using Newtonsoft.Json;

namespace Simulation.Domain.Entities
{
    public class ContributionRequest
    {
        [JsonProperty("name")]
        public string? Name { get; set; }
        [JsonProperty("nominal")]
        public bool? Nominal { get; set; }
        [JsonProperty("values")]
        public Dictionary<int, float>? Values { get; set; }
        [JsonProperty("portfolio")]
        public int? Portfolio { get; set; }

        ContributionRequest() { }
    }
}
